internal/services/containers/container_registry_credential_set_resource_test.go

```diff
@@ -0,0 +1,173 @@
+	data := acceptance.BuildTestData(t, "azurerm_container_registry_credential_set", "test")
+	r := ContainerRegistryCredentialSetResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccContainerRegistryCredentialSet_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_container_registry_credential_set", "test")
+	r := ContainerRegistryCredentialSetResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		{
+			Config:      r.requiresImport(data),
+			ExpectError: acceptance.RequiresImportError("azurerm_container_registry_credential_set"),
+		},
+	})
+}
+
+func TestAccContainerRegistryCredentialSet_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_container_registry_credential_set", "test")
+	r := ContainerRegistryCredentialSetResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (ContainerRegistryCredentialSetResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := credentialsets.ParseCredentialSetID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.Containers.CredentialSetsClient.Get(ctx, *id)
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	return pointer.To(resp.Model != nil), nil
+}
+
+func (ContainerRegistryCredentialSetResource) basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "accTestRG-acr-credetial-set-%d"
+  location = "%s"
+}
+
+resource "azurerm_container_registry" "test" {
+  name                = "testacccr%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  sku                 = "Basic"
+}
+
+resource "azurerm_container_registry_credential_set" "test" {
+  name                  = "testacc-acr-credential-set-%d"
+  container_registry_id = azurerm_container_registry.test.id
+  login_server          = "docker.io"
+  identity {
+    type = "SystemAssigned"
+  }
+  authentication_credentials {
+    username_secret_id = "https://example-keyvault.vault.azure.net/secrets/acr-cs-user-name"
+    password_secret_id = "https://example-keyvault.vault.azure.net/secrets/acr-cs-user-password"
```

Comment:
Running the test locally fails because this doesn't exist, we can you add a test template that creates the dependencies for this acceptance test?